### PR TITLE
refactor: split ConversationViewModel into multiple smaller ViewModels

### PR DESCRIPTION
**Pull request by vitorhugods**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- `ConversationViewModel` is ridiculously huge, hard to maintain and there's no clear separation of anything.
- Testing it is even harder and a full test class would be even huger.
- I need to add more logic (regarding pagination), and I couldn't live with myself if I'd make this class larger.

### Solutions

Split it into 4 ViewModels:
- `ConversationInfoViewModel`, responsible for getting the name, and type of conversation, navigating to conversation details.
- `ConversationMessagesViewModel`, responsible for reading messages in a conversation, including assets.
- `ConversationCallViewModel`, responsible for getting current Call data for a conversation. If there's an active call or not, etc.
- `MessageComposerViewModel`, responsible for composing new messages.

> **Note**: This split into smaller ViewModels is not yet standardised in our codebase, and is something to be discussed in the next collective. However, during last week's discussion with the team, it was majorly seen as something positive and even the alternative approach suggested by @trOnk12 consists in splitting the logic into these smaller chunks and this PR could be easily converted into that, if it flies.

### Testing

- Manually tested all the Conversation flows I could think of.
- _W.I.P._ tests still need to be moved around

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-923/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-923/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-923/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4316453)(4316453)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4316453)(4316453)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-923/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4347545)(4347545)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4347545)(4347545)-dev-release.apk)

**Review by MohamadJaara — APPROVED** [review]
lgtm 🚀

**Review by gongracr — APPROVED** [review]
Massive effort refactoring and making all this code more digestible. I left a tiny renaming suggestion but otherwise looking clean and sharp 🎩 ✨ 👏

**Review by trOnk12 — APPROVED** [review]
Great job and a nice step into cleaner code

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-923/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4349544)(4349544)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4349544)(4349544)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/info/ConversationInfoViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.data.id.QualifiedIdMapper
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.feature.conversation.ObserveConversationDetailsUseCase
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class ConversationInfoViewModel @Inject constructor(
```
[the remark this answers is not in the document]
Comment (vitorhugods) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/info/ConversationInfoViewModel.kt:34]: I missed this comment before merging the PR 🫥

Indeed, I think Metadata would be better.

**Pull request merged**